dawn/src/dawn/Optimizer/PassInlining.cpp

```diff
@@ -155,9 +156,11 @@ class Inliner : public iir::ASTVisitor {
 
       // Register the variable
       metadata_.addAccessIDNamePair(AccessID, returnVarName);
-      metadata_.addStmtToAccessID(newStmt, AccessID);
+      newStmt->getData<iir::VarDeclStmtData>().AccessID = std::make_optional(AccessID);
       // TODO recheck this
-      metadata_.insertExprToAccessID(newExpr_, AccessID);
+      std::dynamic_pointer_cast<iir::VarAccessExpr>(newExpr_)
+          ->getData<iir::IIRAccessExprData>()
```

Comment:
Mostly is something like `node->getData<DataType>().MemberField...` ... we could have helper functions for each member field of each data structure, so it would become `getMemberField(node)` ... but wouldn't it pollute the `dawn::iir` namespace with functions?

Reply:
Hmm yes, we can also postpone this decision and ask others for their opinion. I am not strongly against global functions, especially if we are careful with namespaces. A global function might be a better abstraction in this case, because in this function chain there is significant potential that something changes in the future and we need to go through all of them.